frontend/src/main.ts

```diff
@@ -60,10 +61,74 @@ function defineCustomElements() {
   customElements.define("svelte-component", SvelteCustomElement);
 }
 
+interface ExtensionModule {
+  onExtensionInit?: (this: ExtensionModule) => void;
+  onExtensionPageLoad?: (this: ExtensionModule) => void;
+}
+
+declare global {
+  interface Window {
+    extension?: ExtensionModule;
+  }
+}
+
+/**
+ * Class to wrap an individual extension module and handle calling module functions
+ */
+class Extension {
+  extension_name: string;
+
+  extension_module: ExtensionModule;
+
+  constructor(extension_name: string, extension_module: ExtensionModule) {
+    this.extension_name = extension_name;
+    this.extension_module = extension_module;
+  }
+
+  initExtension(): void {
+    this.tryCall("onExtensionInit", []);
+  }
+
+  onPageLoad(): void {
+    this.tryCall("onExtensionPageLoad", []);
+  }
+
+  private tryCall<
```

Comment:
I'm also planning to get rid of this tryCall abstraction. The TypeScript trickery is more complicated than it's worth I think, and I don't think that it actually works properly if the functions have any arguments.